src/de.rs

```diff
@@ -1,36 +1,31 @@
+use crate::{
+    sealed::binary_heap::Kind as BinaryHeapKind, BinaryHeap, IndexMap, IndexSet, LinearMap, String,
+    Vec,
+};
 use core::{fmt, marker::PhantomData};
-
-use generic_array::{typenum::PowerOfTwo, ArrayLength};
 use hash32::{BuildHasherDefault, Hash, Hasher};
 use serde::de::{self, Deserialize, Deserializer, Error, MapAccess, SeqAccess};
 
-use crate::{
-    indexmap::{Bucket, Pos},
-    sealed::binary_heap::Kind as BinaryHeapKind,
-    BinaryHeap, IndexMap, IndexSet, LinearMap, String, Vec,
-};
-
 // Sequential containers
 
-impl<'de, T, N, KIND> Deserialize<'de> for BinaryHeap<T, N, KIND>
+impl<'de, T, KIND, const N: usize> Deserialize<'de> for BinaryHeap<T, KIND, N>
```

Comment:
Is it okay to put `N` after `KIND`? This PR is a breaking change however it will take more effort for users to upgrade to a new version with const generics.

Reply:
const generics can be passed only after lifetimes and after regular generics by design

Reply:
Indeed, we wanted to not do it but const generics forces it. Same with the default type parameter not working when using const generics :/

Reply:
https://github.com/rust-lang/rust/pull/75384

Reply:
Ah I was referring to default type parameters note working before the const generics parameter.